### ISSUE TITLE
SC1075 (else if -> elif): Trigger iff body is a single statement

### ISSUE THREAD

**Issue by andsens** [koalaman/shellcheck#1584]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

if true; then
  :
else if true; then
    :
  fi
  if true; then
    :
  fi
fi
```

#### Here's what shellcheck currently says:
```
Line 5:
else if true; then
^-- SC1075: Use 'elif' instead of 'else if' (or put 'if' on new line if nesting).
```

#### Here's what I wanted or expected to see:

I would expect this not to be an error because in this situation an `elif` would change the semantics of this script. Currently the wiki states the following:
> `else if` is a valid (though confusing) way of nesting an `if` statement in a parent's `else`. If this is your intention, please use canonical formatting and put a linefeed between `else` and `if`.

Which to me seems like kind of a "documentation escape hatch". Granted, I am running into this problem only because I am minifying code, but this is the only error I get after minification so it leaves me hopeful that there won't be any other syntax-dependent rules after this one is fixed.

Conversely, the following should trigger the rule, but does not:
```
#!/usr/bin/env bash

if true; then
  :
else
  if true; then
    :
  fi
fi
```

**Comment by koalaman** [comment]
You're right. You're using `else if` in a correct and valid way, and ShellCheck's warning doesn't apply.

It's always frustrating when ShellCheck triggers on correct code. Sorry about that. In this case I hope you're willing to bear with it for the sake of the newbies though.

When they make the rather understandable mistake of writing `if true; then :; else if true; then :; fi` , then without this warning, both shells and ShellCheck would be complaining about a missing `fi` even though it's clearly there. The hope is that the warning will be more helpful for them than it's annoying to you.

There are a few others warnings that may trigger when trying to minimize a script, such as `var= foo` vs `var="" foo` and `> file` vs `: > file`, and the reasoning there is similar.

The fact that it doesn't trigger with a linefeed is intentional: `else if` is basically universally written with a single space between them in languages that support it, so `else\nif` is a good indication that you don't consider them a unit. These are obviously all just heuristics, trying to strike an acceptable balance between signal and noise.

**Comment by andsens** [comment]
Ah, ok fair enough. That's a perfectly valid reason and the trade-off benefits the groups that would be helped the most. Thanks for taking the time to write an in-depth explanation. Big fan of this project :-)
